### ISSUE TITLE
Unnecessary dependency -lnsl breaks build on CentOS 8

### ISSUE THREAD

**Issue by gitcnd** [perl5-dbi/DBD-Oracle#120]
I removed -lnsl and it builds fine - well - builds and works, with a lot of warnings (some look important), but those are not nsl related.

Does DBD even use NIS?  Maybe that can be removed on all platforms?

##########
The problem:
rm -f blib/arch/auto/DBD/Oracle/Oracle.so
LD_RUN_PATH="/opt/oracle/product/19c/dbhome_1/lib:/lib64" gcc  -lpthread -shared -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -L/usr/local/lib -fstack-protector-strong  Oracle.o dbdimp.o oci8.o  -o blib/arch/auto/DBD/Oracle/Oracle.so  \
   -L/opt/oracle/product/19c/dbhome_1/lib -lclntsh -ldl -lm -lpthread -lnsl -lirc -limf -lirc -lrt -laio -lresolv -lsvml   \
  
/usr/bin/ld: cannot find -lnsl
collect2: error: ld returned 1 exit status
make: *** [Makefile:526: blib/arch/auto/DBD/Oracle/Oracle.so] Error 1
  MJEVANS/DBD-Oracle-1.80.tar.gz
  /usr/bin/make -- NOT OK
Failed during this command:
 MJEVANS/DBD-Oracle-1.80.tar.gz               : make NO
##########

The solution - NOTE - includes a lot of warnings, including more than a few that probably need attention (buffer overflow things).

DBD-Oracle-1.80-2]# make
Skip blib/lib/DBD/Oracle/Troubleshooting/Linux.pod (unchanged)
Skip blib/arch/auto/DBD/Oracle/ocitrace.h (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting/Hpux.pod (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting/Win64.pod (unchanged)
Skip blib/arch/auto/DBD/Oracle/Oracle.h (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting.pod (unchanged)
Skip blib/lib/DBD/Oracle/GetInfo.pm (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting/Win32.pod (unchanged)
Skip blib/lib/DBD/Oracle/Object.pm (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting/Macos.pod (unchanged)
Skip blib/arch/auto/DBD/Oracle/dbdimp.h (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting/Vms.pod (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting/Aix.pod (unchanged)
cp mk.pm blib/arch/auto/DBD/Oracle/mk.pm
Skip blib/lib/DBD/Oracle/Troubleshooting/Sun.pod (unchanged)
Skip blib/lib/DBD/Oracle/Troubleshooting/Cygwin.pod (unchanged)
Skip blib/lib/DBD/Oracle.pm (unchanged)
Running Mkbootstrap for Oracle ()
chmod 644 "Oracle.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Oracle.bs blib/arch/auto/DBD/Oracle/Oracle.bs 644
gcc -c  -I/opt/oracle/product/19c/dbhome_1/rdbms/public -I/usr/local/lib64/perl5/auto/DBI -D_REENTRANT -D_GNU_SOURCE -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fwrapv -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -g   -DVERSION=\"1.80\" -DXS_VERSION=\"1.80\" -fPIC "-I/usr/lib64/perl5/CORE"  -Wall -Wno-comment -DUTF8_SUPPORT -DORA_OCI_VERSION=\"19.3.0.0\" -DORA_OCI_102 -DORA_OCI_112 Oracle.c
In file included from dbdimp.h:439,
                 from Oracle.h:52,
                 from Oracle.xs:1:
Oracle.xs: In function ‘XS_DBD__Oracle__db_ora_db_startup’:
Oracle.xs:324:82: warning: macro expands to multiple statements [-Wmultistatement-macros]
  if (admhp) OCIHandleFree_log_stat(imp_dbh, (dvoid*)admhp, (ub4)OCI_HTYPE_ADMIN, status);
                                                                                  ^~~~~~
ocitrace.h:411:2: note: in definition of macro ‘OCIHandleFree_log_stat’
  stat=OCIHandleFree( (hp), (t));    \
  ^~~~
Oracle.xs:324:2: note: some parts of macro expansion are not guarded by this ‘if’ clause
  if (admhp) OCIHandleFree_log_stat(imp_dbh, (dvoid*)admhp, (ub4)OCI_HTYPE_ADMIN, status);
  ^~
gcc -c  -I/opt/oracle/product/19c/dbhome_1/rdbms/public -I/usr/local/lib64/perl5/auto/DBI -D_REENTRANT -D_GNU_SOURCE -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fwrapv -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -g   -DVERSION=\"1.80\" -DXS_VERSION=\"1.80\" -fPIC "-I/usr/lib64/perl5/CORE"  -Wall -Wno-comment -DUTF8_SUPPORT -DORA_OCI_VERSION=\"19.3.0.0\" -DORA_OCI_102 -DORA_OCI_112 dbdimp.c
In file included from dbdimp.h:439,
                 from Oracle.h:52,
                 from dbdimp.c:21:
dbdimp.c: In function ‘ora_db_login6’:
dbdimp.c:923:57: warning: macro expands to multiple statements [-Wmultistatement-macros]
        (ub4) OCI_ATTR_CONNECTION_CLASS, imp_dbh->errhp, status);
                                                         ^~~~~~
ocitrace.h:325:2: note: in definition of macro ‘OCIAttrSet_log_stat’
  stat=OCIAttrSet(th,ht,ah,s1,a,eh);    \
  ^~~~
dbdimp.c:920:4: note: some parts of macro expansion are not guarded by this ‘if’ clause
    if (imp_dbh->pool_class) /*pool_class may or may not be used */
    ^~
dbdimp.c: In function ‘ora_bind_ph’:
dbdimp.c:3401:4: warning: increment of a boolean expression [-Wbool-operation]
    ++imp_sth->has_inout_params;
    ^~
In file included from dbdimp.h:439,
                 from Oracle.h:52,
                 from dbdimp.c:21:
dbdimp.c: In function ‘ora_free_phs_contents’:
ocitrace.h:380:2: warning: macro expands to multiple statements [-Wmultistatement-macros]
  if (DBD_OCI_TRACEON(impxxh)) PerlIO_printf(DBD_OCI_TRACEFP(impxxh), \
  ^~
dbdimp.c:4395:9: note: in expansion of macro ‘OCIDescriptorFree_log’
         OCIDescriptorFree_log(imp_sth, phs->desc_h, phs->desc_t);
         ^~~~~~~~~~~~~~~~~~~~~
dbdimp.c:4394:2: note: some parts of macro expansion are not guarded by this ‘if’ clause
  if (phs->desc_h)
  ^~
dbdimp.c: In function ‘dump_env_to_trace’:
dbdimp.c:4764:29: warning: comparison between pointer and zero character constant [-Wpointer-compare]
  } while ((char*)environ[i] != '\0');
                             ^~
dbdimp.c:4764:11: note: did you mean to dereference the pointer?
  } while ((char*)environ[i] != '\0');
           ^
gcc -c  -I/opt/oracle/product/19c/dbhome_1/rdbms/public -I/usr/local/lib64/perl5/auto/DBI -D_REENTRANT -D_GNU_SOURCE -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fwrapv -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -g   -DVERSION=\"1.80\" -DXS_VERSION=\"1.80\" -fPIC "-I/usr/lib64/perl5/CORE"  -Wall -Wno-comment -DUTF8_SUPPORT -DORA_OCI_VERSION=\"19.3.0.0\" -DORA_OCI_102 -DORA_OCI_112 oci8.c
oci8.c:18: warning: "sv_set_undef" redefined
 #define sv_set_undef(sv) if (SvROK(sv)) sv_unref(sv); else SvOK_off(sv)
 
In file included from /usr/lib64/perl5/CORE/perl.h:5644,
                 from /usr/local/lib64/perl5/auto/DBI/DBIXS.h:23,
                 from Oracle.h:50,
                 from oci8.c:12:
/usr/lib64/perl5/CORE/embed.h:673: note: this is the location of the previous definition
 #define sv_set_undef(a)  Perl_sv_set_undef(aTHX_ a)
 
In file included from dbdimp.h:439,
                 from Oracle.h:52,
                 from oci8.c:12:
oci8.c: In function ‘init_lob_refetch’:
ocitrace.h:380:2: warning: macro expands to multiple statements [-Wmultistatement-macros]
  if (DBD_OCI_TRACEON(impxxh)) PerlIO_printf(DBD_OCI_TRACEFP(impxxh), \
  ^~
oci8.c:4571:3: note: in expansion of macro ‘OCIDescriptorFree_log’
   OCIDescriptorFree_log(imp_sth, colhd, OCI_DTYPE_PARAM);
   ^~~~~~~~~~~~~~~~~~~~~
oci8.c:4570:2: note: some parts of macro expansion are not guarded by this ‘if’ clause
  if (colhd)
  ^~
In file included from dbdimp.h:439,
                 from Oracle.h:52,
                 from oci8.c:12:
oci8.c: In function ‘ora_free_lob_refetch’:
ocitrace.h:380:2: warning: macro expands to multiple statements [-Wmultistatement-macros]
  if (DBD_OCI_TRACEON(impxxh)) PerlIO_printf(DBD_OCI_TRACEFP(impxxh), \
  ^~
oci8.c:4911:3: note: in expansion of macro ‘OCIDescriptorFree_log’
   OCIDescriptorFree_log(imp_sth, lr->rowid, OCI_DTYPE_ROWID);
   ^~~~~~~~~~~~~~~~~~~~~
oci8.c:4910:2: note: some parts of macro expansion are not guarded by this ‘if’ clause
  if (lr->rowid)
  ^~
oci8.c: In function ‘get_attr_val’:
oci8.c:2337:25: warning: ‘%02d’ directive writing between 2 and 4 bytes into a region of size 2 [-Wformat-overflow=]
     sprintf(s_tz_hour," %02d",tz_hour);
                         ^~~~
oci8.c:2337:23: note: directive argument in the range [-128, 127]
     sprintf(s_tz_hour," %02d",tz_hour);
                       ^~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/lib64/perl5/CORE/perlio.h:41,
                 from /usr/lib64/perl5/CORE/iperlsys.h:51,
                 from /usr/lib64/perl5/CORE/perl.h:3880,
                 from /usr/local/lib64/perl5/auto/DBI/DBIXS.h:23,
                 from Oracle.h:50,
                 from oci8.c:12:
/usr/include/bits/stdio2.h:36:10: note: ‘__builtin___sprintf_chk’ output between 4 and 6 bytes into a destination of size 3
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
oci8.c:2335:25: warning: ‘%03d’ directive writing between 3 and 4 bytes into a region of size 2 [-Wformat-overflow=]
     sprintf(s_tz_hour," %03d",tz_hour);
                         ^~~~
oci8.c:2335:23: note: directive argument in the range [-128, 127]
     sprintf(s_tz_hour," %03d",tz_hour);
                       ^~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/lib64/perl5/CORE/perlio.h:41,
                 from /usr/lib64/perl5/CORE/iperlsys.h:51,
                 from /usr/lib64/perl5/CORE/perl.h:3880,
                 from /usr/local/lib64/perl5/auto/DBI/DBIXS.h:23,
                 from Oracle.h:50,
                 from oci8.c:12:
/usr/include/bits/stdio2.h:36:10: note: ‘__builtin___sprintf_chk’ output between 5 and 6 bytes into a destination of size 3
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
oci8.c:2340:23: warning: ‘%02d’ directive writing between 2 and 4 bytes into a region of size 2 [-Wformat-overflow=]
    sprintf(s_tz_min,":%02d", tz_minute);
                       ^~~~
oci8.c:2340:21: note: directive argument in the range [-128, 127]
    sprintf(s_tz_min,":%02d", tz_minute);
                     ^~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/lib64/perl5/CORE/perlio.h:41,
                 from /usr/lib64/perl5/CORE/iperlsys.h:51,
                 from /usr/lib64/perl5/CORE/perl.h:3880,
                 from /usr/local/lib64/perl5/auto/DBI/DBIXS.h:23,
                 from Oracle.h:50,
                 from oci8.c:12:
/usr/include/bits/stdio2.h:36:10: note: ‘__builtin___sprintf_chk’ output between 4 and 6 bytes into a destination of size 3
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
rm -f blib/arch/auto/DBD/Oracle/Oracle.so
LD_RUN_PATH="/opt/oracle/product/19c/dbhome_1/lib:/lib64" gcc  -lpthread -shared -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -L/usr/local/lib -fstack-protector-strong  Oracle.o dbdimp.o oci8.o  -o blib/arch/auto/DBD/Oracle/Oracle.so  \
   -L/opt/oracle/product/19c/dbhome_1/lib -lclntsh -ldl -lm -lpthread  -lirc -limf -lirc -lrt -laio -lresolv -lsvml   \
  
chmod 755 blib/arch/auto/DBD/Oracle/Oracle.so
Manifying 13 pod documents
[root@rnw DBD-Oracle-1.80-2]#

**Comment by djzort** [comment]
This is a duplicate of #117 and yes for now its required.

**Comment by gitcnd** [comment]
@djzort  If it was required, it would not build, link and run - which it does just fine without.

I've built and tested it.  What did YOU do?

**Comment by djzort** [comment]
Over the last 18 months I have helped arrange for DBD::Oracle to move from the Pythinian gh repo's to the perl5-dbi repo where it can be maintained. I then set up all the travis CI config and worked to land all the PR's and get releases out, as well as a large number of tidy up diffs and some fixes along the way.

So just that.

But I don't have any access to this repo nor DBD::Oracle permissions on PAUSE. So if there's a problem i am happy to comment on GH issues and test out peoples patches so I can support their inclusion. But have i personally spent any time removing the dep and testing? no i haven't as it isn't impacting me.
